packages/Search/examples/point_clouds/point_clouds.hpp

```diff
@@ -0,0 +1,216 @@
+
+            break;
+        }
+        case 2:
+        {
+            random_points( i ) = {{random(), -half_edge, random()}};
+
+            break;
+        }
+        case 3:
+        {
+            random_points( i ) = {{random(), half_edge, random()}};
+
+            break;
+        }
+        case 4:
+        {
+            random_points( i ) = {{random(), random(), -half_edge}};
+
+            break;
+        }
+        case 5:
+        {
+            random_points( i ) = {{random(), random(), half_edge}};
+
+            break;
+        }
+        default:
+        {
+            throw std::runtime_error( "Your compiler is broken" );
+        }
+        }
+    }
+}
+
+template <typename Layout, typename DeviceType>
+void filledSphereCloud(
+    double const radius,
+    Kokkos::View<DataTransferKit::Point *, Layout, DeviceType> random_points )
+{
+    static_assert(
+        Kokkos::Impl::MemorySpaceAccess<
+            Kokkos::HostSpace, typename DeviceType::memory_space>::accessible,
+        "The View should be accessible on the Host" );
+    std::default_random_engine generator;
+
+    std::uniform_real_distribution<double> distribution( -radius, radius );
+    auto random = [&distribution, &generator]() {
+        return distribution( generator );
+    };
+
+    unsigned int const n = random_points.extent( 0 );
+    for ( unsigned int i = 0; i < n; ++i )
+    {
+        bool point_accepted = false;
+        while ( !point_accepted )
+        {
+            double const x = random();
+            double const y = random();
+            double const z = random();
+
+            // Only accept points that are in the sphere
+            if ( std::sqrt( x * x + y * y + z * z ) <= radius )
+            {
+                random_points( i ) = {{x, y, z}};
+                point_accepted = true;
+            }
+        }
+    }
+}
+
+template <typename Layout, typename DeviceType>
+void hollowSphereCloud(
+    double const radius,
+    Kokkos::View<DataTransferKit::Point *, Layout, DeviceType> random_points )
+{
+    static_assert(
+        Kokkos::Impl::MemorySpaceAccess<
+            Kokkos::HostSpace, typename DeviceType::memory_space>::accessible,
+        "The View should be accessible on the Host" );
+    std::default_random_engine generator;
+
+    std::uniform_real_distribution<double> distribution( -1., 1. );
+    auto random = [&distribution, &generator]() {
+        return distribution( generator );
+    };
+
+    unsigned int const n = random_points.extent( 0 );
+    for ( unsigned int i = 0; i < n; ++i )
+    {
+        double const x = random();
+        double const y = random();
+        double const z = random();
+        double const norm = std::sqrt( x * x + y * y + z * z );
+
+        if ( norm > 1e-15 )
+            random_points( i ) = {
+                {radius * x / norm, radius * y / norm, radius * z / norm}};
+        else
+            random_points( i ) = {{0., 0., 0.}};
```

Comment:
Did you actually get a division by zero without that else-clause?

Reply:
I don't think I ever tried without the clause

Reply:
I'd be tempted to remove the if- or reject the point like you did for the "filled" sphere.

Reply:
I've removed the if else